fdbserver/ptxn/Serializer.h

```diff
@@ -43,46 +42,41 @@ size_t getSerializedBytes() {
 	return value;
 }
 
-/**
- * @class HeaderedItemsSerializerBase
- *
- * Encode objects in the format
- *
- *  | Header | Item | Item | ... |
- *
- * The Header MUST have a fixed size when serialized.
- */
+// Encode objects in the format
+//
+//  | Header | Item | Item | ... |
+//
+// The Header MUST have a fixed size when serialized.
 template <typename Header, typename Item>
-class HeaderedItemsSerializerBase {
+class HeaderedItemsSerializer {
 	BinaryWriter writer;
 
-	/// If true, no more items should be written to the writer
+	// If true, no more items should be written to the writer
 	bool itemComplete = false;
 
-	/// Number of items
+	// Number of items
 	size_t numItems = 0;
 
 public:
 	using header_t = Header;
 	using item_t = Item;
 
-	HeaderedItemsSerializerBase() : writer(BinaryWriter(AssumeVersion(g_network->protocolVersion()))) {
-		writer << header_t();
-	}
+	HeaderedItemsSerializer() : writer(AssumeVersion(g_network->protocolVersion())) { writer << header_t(); }
```

Comment:
For protocol version, I think the custom is to define one in `flow/ProfocolVersion.h`, such as
```
PROTOCOL_VERSION_FEATURE(0x0FDB00B071010000LL, PartitionedTransaction);
```
, assuming this is for 7.1 version. Then you can use
`writer(IncludeVersion(ProtocolVersion::withPartitionedTransaction()))`